src/lib.rs

```diff
@@ -46,27 +46,15 @@ compile_error!("at least one of the `std` or `no-std` features must be enabled")
 #[cfg(feature = "no-std")]
 #[macro_use]
 extern crate alloc;
-#[cfg(feature = "no-std")]
-extern crate core2;
-
-#[cfg(any(feature = "std", test))]
-extern crate core; // for Rust 1.29 and no-std tests
 
 // Re-exported dependencies.
 #[macro_use] pub extern crate bitcoin_hashes as hashes;
 pub extern crate secp256k1;
 pub extern crate bech32;
 
-#[cfg(feature = "no-std")]
-extern crate hashbrown;
-
 #[cfg(feature = "base64")] pub extern crate base64;
 
-#[cfg(feature="bitcoinconsensus")] extern crate bitcoinconsensus;
 #[cfg(feature = "serde")] #[macro_use] extern crate serde;
-#[cfg(all(test, feature = "serde"))] extern crate serde_json;
-#[cfg(all(test, feature = "serde"))] extern crate serde_test;
-#[cfg(all(test, feature = "serde"))] extern crate bincode;
 #[cfg(all(test, feature = "unstable"))] extern crate test;
```

Comment:
Why do we need to keep this last line?

Reply:
`test` is a special crate - see commit message. But TBH I'm not deeply knowledgeable about it. I based my change on compiler error message.